src/main/java/com/cs3365/rpgdapi/weapons/WeaponsController.java

```diff
@@ -64,4 +59,40 @@ public ResponseEntity get(@PathVariable("weaponID") UUID identifier)
                                          e.getMessage()));
         }
     }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<String> put(@RequestBody Weapon requestBody,
+                                      @PathVariable UUID identity) throws WeaponsException {
+        try {
+            requestBody.validate();
+        } catch(WeaponsException e) {
+            return ResponseEntity.badRequest().body(e.getMessage());
+        }
+
+        WeaponEntity weapon = weaponsService.findWeapon(identity);
+        if (weapon == null) {
+            return ResponseEntity.notFound().build();
+        }
+
+        weapon.setName(requestBody.getName());
+        weapon.setType(requestBody.getType());
+        weapon.setDescription(requestBody.getDescription());
+        weapon.setAttackPower(requestBody.getAttackPower());
+        weapon.setAttackType(requestBody.getAttackType());
+        weapon.setSpecialAbility(requestBody.getSpecialAbility());
+        weapon.setWeight(requestBody.getWeight());
+
+        try {
+            weaponsService.updateWeapon(weapon);
```

Comment:
All of this code should live in the `WeaponsServiceImpl` class. This is all business logic, and all services in this application would be considered part of the business logic layer. The controller classes are the presentation layer.

For line 67 you could argue it is part of the business logic layer, but typically I've seen request validation handled in the presentation layer in several enterprise applications. It's one of those things that's at your discretion, but I think it's good here.

Line 86 should be replaced by something like `WeaponEntity updatedWeapon = weaponsService.updateWeapon(identity, weapon)`. More below about the method signature.